packages/libs/app-sdk/src/jwt/typeGuards.ts

```diff
@@ -1,36 +1,71 @@
-import { JWT_ERROR } from 'did-jwt';
-
-import type { JWTDecoded, VincentJWT, VincentJWTAppSpecific } from './types';
+import type {
+  VincentJWTPlatformUser,
+  VincentJWTAppUser,
+  VincentJWTDelegatee,
+  AnyVincentJWT,
+  DecodedJWT,
+} from './types';
 
+import { JWT_ERROR, VINCENT_JWT_API_VERSION } from './constants';
 import { isDefinedObject } from './core/utils/index';
 
-/** Use this typeguard function to identify if the JWT is appId specific and make subsequent type-safe
- * references into the payload of the JWT
- */
-export function isAppSpecificJWT(decodedJWT: VincentJWT): decodedJWT is VincentJWTAppSpecific {
-  return decodedJWT.payload.app && decodedJWT.payload.app.id;
+/**
+ * Check if a decoded JWT is an app-specific JWT (role === 'app-user')
+ *
+ * @category API > Type Guards
+ * */
+export function isAppUser(decodedJWT: DecodedJWT): decodedJWT is VincentJWTAppUser {
+  return decodedJWT.payload?.role === 'app-user';
 }
 
-/** Use this typeguard function to identify if the JWT is a general authentication JWT that has no specific app target */
-export function isGeneralJWT(decodedJWT: VincentJWT): decodedJWT is VincentJWT {
-  return !isAppSpecificJWT(decodedJWT);
+/** Check if a decoded JWT is a general platform-user JWT
+ *
+ * @category API > Type Guards
+ * */
+export function isPlatformUser(decodedJWT: DecodedJWT): decodedJWT is VincentJWTPlatformUser {
+  return decodedJWT.payload?.role === 'platform-user';
+}
+
+/** Check if a decoded JWT is a delegatee token (role === 'app-delegatee')
+ *
+ * @category API > Type Guards
+ * */
+export function isDelegateee(decodedJWT: DecodedJWT): decodedJWT is VincentJWTDelegatee {
```

Comment:
I think there's too many eee's in delegatee here

Reply:
Removed extra "e"